frontend/e2e/pages/metrics/report-step.ts

```diff
@@ -339,27 +406,98 @@ export class ReportStep {
     await expect(this.exportPipelineDataButton).toBeEnabled();
   }
 
-  async checkBoardMetricsDetails(boardDetailType: ProjectCreationType, csvCompareLines: number) {
+  async checkBoardMetricsDetailsForMultipleRanges({
+    projectCreationType,
+    csvCompareLines,
+    velocityData,
+    cycleTimeData,
+    classificationData,
+    reworkData,
+  }: {
+    projectCreationType: ProjectCreationType;
+    csvCompareLines: ICsvComparedLines;
+    velocityData: IBoardMetricsDetailItem[][];
+    cycleTimeData: IBoardCycletimeDetailItem[][];
+    classificationData: IBoardClassificationDetailItem[][];
+    reworkData: IBoardMetricsDetailItem[][];
+  }) {
     await this.showMoreLinks.first().click();
     if (
-      boardDetailType === ProjectCreationType.IMPORT_PROJECT_FROM_FILE ||
-      boardDetailType === ProjectCreationType.CREATE_A_NEW_PROJECT
+      projectCreationType === ProjectCreationType.IMPORT_PROJECT_FROM_FILE ||
+      projectCreationType === ProjectCreationType.CREATE_A_NEW_PROJECT
     ) {
-      await this.checkBoardMetricsReportReportDetail();
+      await this.checkVelocityDetailForMultipleRanges(velocityData);
+      await this.checkCycleTimeDetailForMultipleRanges(cycleTimeData);
+      await this.checkClassificationDetailForMultipleRanges(classificationData);
+      await this.checkReworkDetailForMultipleRanges(reworkData);
     } else {
       throw Error('The board detail type is not correct, please give a correct one.');
     }
 
-    await downloadFileAndCheck(this.page, this.exportBoardData, 'boardData.csv', async (fileDataString) => {
-      const localCsvFile = fs.readFileSync(path.resolve(__dirname, '../../fixtures/create-new/board-data.csv'));
-      const localCsv = parse(localCsvFile, { to: csvCompareLines });
-      const downloadCsv = parse(fileDataString, { to: csvCompareLines });
-
-      expect(localCsv).toStrictEqual(downloadCsv);
+    await this.downloadFileAndCheckForMultipleRanges({
+      trigger: this.exportBoardData,
+      rangeCount: csvCompareLines.length,
+      csvCompareLines,
     });
     await this.backButton.click();
   }
 
+  async downloadFileAndCheckForMultipleRanges({
+    trigger,
+    rangeCount,
+    csvCompareLines,
+  }: {
+    trigger: Locator;
+    rangeCount: number;
+    csvCompareLines?: ICsvComparedLines;
+  }) {
+    const isNeedToCompareCsvLines = csvCompareLines !== undefined;
+    if (isNeedToCompareCsvLines && rangeCount !== csvCompareLines.length) {
+      throw new Error('given time ranges count and csvCompareLines count are not equal, please check your fixture');
+    }
```

Comment:
Why `throw new Error` in the e2e test case?
there is no throw/if in the the e2e test sceniro

Reply:
To ensure the range count is equal to `compareCsvLines` if `compareCsvLines` presents.

There IS already some `throw Error` in the e2e test(`e2e/pages/metrics/report-step.ts`)

Reply:
have you fix it???